### PR TITLE
Beginning EthSignTransaction implementation

### PR DESCRIPTION
**Pull request by mtleliever**
- Implemented EthereumAppSignTransactionResponse to match the output data of the ledger transaction signature. 
- Implemented arguments for the EthereumAppSignTransactionRequest.

Successful attempts to sign a transaction were made in the other branch https://github.com/ThatSlyGuy/Ledger.Net/tree/SignEthereumTransaction-SuccessfulTests

https://github.com/LedgerHQ/ledger-app-eth/blob/master/doc/ethapp.asc 
See "SIGN ETH TRANSACTION" section.

### CODE REVIEW

File: src/Ledger.Net/Requests/Concrete/EthereumAppSignTransactionRequest.cs [hunk 1]
```diff
@@ -2,20 +2,28 @@
...
+        // Javascript implementation: https://github.com/LedgerHQ/ledgerjs/blob/master/packages/hw-app-eth/src/Eth.js#L147
+        // Foreach function uses 0x00 on the first index of the data array and 0x80 on the rest.
+        // Foreach implementation is moved to the utils.js file: https://github.com/LedgerHQ/ledgerjs/blob/master/packages/hw-app-eth/src/utils.js#L58
+
+        private const byte P1_FIRST_TRANSACTION_BLOCK = 0x00;
+        private const byte P1_SUBSEQUENT_TRANSACTION_BLOCK = 0x80;
+
         #region Public Overrides
-        public override byte Argument1 => 0;
+        public override byte Argument1 => FirstTransactionDataBlock ? P1_FIRST_TRANSACTION_BLOCK : P1_SUBSEQUENT_TRANSACTION_BLOCK;
```
[the remark this answers is not in the document]
Comment (ThatSlyGuy) [reply src/Ledger.Net/Requests/Concrete/EthereumAppSignTransactionRequest.cs]: https://github.com/LedgerHQ/ledgerjs/blob/master/packages/hw-app-eth/src/Eth.js#L147

[the remark this answers is not in the document]
Comment (ThatSlyGuy) [reply src/Ledger.Net/Requests/Concrete/EthereumAppSignTransactionRequest.cs]: For failed implementation attempt see EthHelpers, LedgerManager, and LedgerTests in https://github.com/ThatSlyGuy/Ledger.Net/tree/SignEthereumTransaction-FailedTests

[the remark this answers is not in the document]
Comment (ThatSlyGuy) [reply src/Ledger.Net/Requests/Concrete/EthereumAppSignTransactionRequest.cs]: Successful implementation attempt: https://github.com/ThatSlyGuy/Ledger.Net/tree/SignEthereumTransaction-SuccessfulTests

**Comment by mtleliever** [comment]
Closing pull request since a successful attempt at signing an Ethereum transaction has been made.

**Pull request closed without merging**